src/databricks/labs/dqx/installer/workflow_installer.py

```diff
@@ -487,9 +488,78 @@ def _library_dep_order(library: str):
             case _:
                 return 2
 
+    @staticmethod
+    def _extract_dependency_prefix(requirement: str) -> str | None:
+        """
+        Extract package name prefix from a requirement string.
+
+        Args:
+            requirement: Requirement string (e.g., "databricks-sdk>=0.71")
+
+        Returns:
+            Package name prefix with underscores, or None if invalid.
+        """
+        match = re.match(r'^([a-zA-Z0-9-]+)', requirement)
+        if match:
+            pkg_name = match.group(1)
+            return pkg_name.replace('-', '_')
+        return None
+
+    @staticmethod
+    def _get_fallback_dependencies() -> list[str]:
+        """
+        Get fallback dependency prefixes when metadata is unavailable.
+
+        Returns:
+            List of core dependency prefixes.
+        """
+        return [
+            "databricks_labs_blueprint",
+            "databricks_sdk",
+            "databricks_labs_lsql",
+            "sqlalchemy",
+        ]
+
+    @staticmethod
+    def _get_dependency_prefixes() -> list[str]:
+        """
+        Dynamically retrieve dependency prefixes from package metadata.
+
+        Returns:
+            List of dependency package name prefixes for wheel files.
+        """
+        try:
+            requires = importlib.metadata.requires('databricks-labs-dqx')
+        except importlib.metadata.PackageNotFoundError:
+            logger.warning("databricks-labs-dqx package metadata not found, using fallback dependencies")
+            return WorkflowDeployment._get_fallback_dependencies()
+
+        if not requires:
+            logger.warning("No dependencies found in package metadata")
+            return []
+
+        prefixes = []
+        for req in requires:
+            # Skip optional dependencies (those with markers like extra == 'pii' or 'llm')
```

Comment:
we do support extras in workflows. they are installed by default. If this is not possible, at least a note in the installation should be provided, that extras are not covered.